File: LeetCode/[M] - 238. Product of Array Except Self/solution.py
class Solution:
    def productExceptSelf(self, nums: list[int]) -> list[int]:
        len_nums = len(nums)
        output = [nums[0]] * len_nums
        for i in range(1, len_nums):
            output[i] = output[i - 1] * nums[i]
        product_r = 1
        for j in range(len_nums - 1, 0, -1):
            output[j] = output[j - 1] * product_r
            product_r *= nums[j]
        output[0] = product_r
        return output

File: LeetCode/[M] - 238. Product of Array Except Self/solution.go
package main

func productExceptSelf(nums []int) []int {
	output := make([]int, len(nums))
	output[0] = nums[0]
	for i := 1; i < len(nums); i++ {
		output[i] = output[i-1] * nums[i]
	}
	right := 1
	for j := len(nums) - 1; j > 0; j-- {
		output[j] = output[j-1] * right
		right *= nums[j]
	}
	output[0] = right
	return output
}
